### PR TITLE
Layergroups: attribution & metadatalinks API+rest+getcapabilities

### PR DESCRIPTION
**Pull request by NielsCharlier**
See https://github.com/geoserver/geoserver/wiki/GSIP-135

https://osgeo-org.atlassian.net/browse/GEOS-5888
https://osgeo-org.atlassian.net/browse/GEOS-7313

### CODE REVIEW

**Comment by aaime** [comment]
The copyright header year has not been updated in any of the files. Kevin shared a script to fix the headers in all the files changed in the last commit, look for it in the devel list.

File: src/main/src/main/java/org/geoserver/catalog/impl/LayerGroupInfoImpl.java [hunk 1]
```diff
@@ -388,4 +394,23 @@ public String getPrefixedName() {
     public PublishedType getType() {
         return PublishedType.GROUP;
     }
+
+    @Override
```
Comment (aaime) [inline src/main/src/main/java/org/geoserver/catalog/impl/LayerGroupInfoImpl.java]: Hashcode and equals have not been updated to include the two new fields

File: src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/CapabilitiesTest.java [hunk 2]
```diff
@@ -261,6 +264,41 @@ public void testAttribution() throws Exception {
...
+        info.setType("text/html");
+        info.setMetadataType("FGDC");
+        info.setContent("http://my/metadata/link");
+        lg.getMetadataLinks().add(info);
+        getCatalog().add(lg);
+        
+        try {
+            System.out.println(getAsString("wms?service=WMS&request=getCapabilities&version=1.1.1"));
+            Document doc = getAsDOM("wms?service=WMS&request=getCapabilities&version=1.1.1", true);
+            print(doc);
```
Comment (aaime) [inline src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/CapabilitiesTest.java]: Handy for debugging, but nasty to have at runtime, please comment out this "print"

**Comment by aaime** [comment]
Found some small issues, besides that, looks good.

**Comment by NielsCharlier** [comment]
Is this good to go after fixes?

**Comment by aaime** [comment]
Yep, looks good to me

**Pull request merged**